### DISCUSSION TITLE
Feature request: Allow for use of API Tokens (bypass 2FA)

### DISCUSSION THREAD

**Discussion by kzaoaai** [dougiteixeira/proxmoxve#115] [Feature request] [answered]
I tried to create a token for hass, but it seems like this addon is expecting a username and password.

HA Log:
Config entry 'PROXMOX-M' for proxmoxve integration could not authenticate: Couldn't authenticate user: homeassistant@pve!hass

Proxmox Log:
pvedaemon[1230]: authentication failure; rhost=::ffff:192.xxx.xxx.xx user=homeassistant@pve!hass msg=value 'homeassistant@pve!hass' does not look like a valid user name

![image](https://user-images.githubusercontent.com/39029008/234405002-34f33136-57a0-46f4-931d-badfc4e6c5a2.png)

**Comment by dougiteixeira** [comment]
Authentication via tokens is currently not supported.

I would be grateful to accept a contribution to implement it.

**Comment by dougiteixeira** [comment, marked as answer]
#343
